lib/web/apiserver_test.go

```diff
@@ -2709,6 +2710,136 @@ func TestClusterDatabasesGet(t *testing.T) {
 	}, resp.Items[0])
 }
 
+func TestClusterDatabaseGet(t *testing.T) {
+	env := newWebPack(t, 1)
+	ctx := context.Background()
+
+	proxy := env.proxies[0]
+
+	// Register a database.
+	dbName := "existingdb"
+	db, err := types.NewDatabaseV3(types.Metadata{
+		Name: dbName,
+		Labels: map[string]string{
+			"env": "prod",
+		},
+	}, types.DatabaseSpecV3{
+		Protocol: "test-protocol",
+		URI:      "test-uri",
+	})
+	require.NoError(t, err)
+
+	dbNames := []string{"db1", "db2"}
+	dbUsers := []string{"user1", "user2"}
+
+	err = env.server.Auth().CreateDatabase(context.Background(), db)
+	require.NoError(t, err)
+
+	for _, tt := range []struct {
+		name            string
+		databaseName    string
+		userRoles       func(*testing.T) []types.Role
+		expectedDBUsers []string
+		expectedDBNames []string
+		requireError    require.ErrorAssertionFunc
+	}{
+		{
+			name:         "valid",
+			databaseName: dbName,
+			requireError: require.NoError,
+		},
+		{
+			name:         "notfound",
+			databaseName: "notfound",
+			requireError: func(tt require.TestingT, err error, i ...interface{}) {
+				require.True(tt, trace.IsNotFound(err), "expected a not found error, got %v", err)
+			},
+		},
+		{
+			name:         "notauthorized",
+			databaseName: dbName,
+			userRoles: func(tt *testing.T) []types.Role {
+				role, err := types.NewRole(
+					"myrole",
+					types.RoleSpecV5{
+						Allow: types.RoleConditions{
+							DatabaseLabels: types.Labels{
+								"env": apiutils.Strings{"staging"},
+							},
+						},
+					},
+				)
+				require.NoError(tt, err)
+				return []types.Role{role}
+			},
+			requireError: func(tt require.TestingT, err error, i ...interface{}) {
+				require.True(tt, trace.IsAccessDenied(err), "expected a access denied error, got %v", err)
+			},
+		},
+		{
+			name:         "authorizedDBNamesUsers",
```

Comment:
Can you also add a test case with no databases? This can help with some corner cases.